mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+            )
+        elif self._underlying_provider is AWSBedrockModelProvider.ANTHROPIC:
+            if not prompt.startswith("Human: "):
+                prompt = f"Human: {prompt}"
+            if not prompt.endswith("\n\nAssistant:"):
+                prompt = f"{prompt}\n\nAssistant:"
+
+            preped["prompt"] = prompt
+
+            # rename and ensure
+            preped["max_tokens_to_sample"] = max_tokens = payload.pop(
+                "max_tokens", MLFLOW_AI_GATEWAY_ANTHROPIC_DEFAULT_MAX_TOKENS
+            )
+
+            preped["max_tokens_to_sample"] = min(preped["max_tokens_to_sample"], 8191)
+            if max_tokens > MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS:
+                raise HTTPException(
+                    status_code=422,
+                    detail="Invalid value for max_tokens: cannot exceed "
+                    f"{MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS}.",
+                )
+
+            preped = {**preped, **rename_payload_keys(payload, {"stop": "stop_sequences"})}
+            if "stop_sequences" not in preped:
+                preped["stop_sequences"] = []
+
+            if "\n\nHuman:" not in preped["stop_sequences"]:
+                preped["stop_sequences"].append("\n\nHuman:")
+        elif self._underlying_provider is AWSBedrockModelProvider.AI21:
+            preped = {
+                "prompt": prompt,
+                **rename_payload_keys(
+                    payload,
+                    {
+                        "stop": "stopSequences",
+                        "candidate_count": "numResults",
+                        "max_tokens": "maxTokens",
+                    },
+                ),
+            }
+
+        elif self._underlying_provider is AWSBedrockModelProvider.COHERE:
+            preped = {
+                "prompt": prompt,
+                **rename_payload_keys(
+                    payload,
+                    {
+                        "stop": "stop_sequences",
+                        "candidate_count": "num_generations",
+                    },
+                ),
+            }
+            # The range of Cohere's temperature is 0-5, but ours is 0-1, so we scale it.
+            if "temperature" in preped:
+                preped["temperature"] *= 5
+        else:
+            raise HTTPException(
+                status_code=422,
+                detail=f"Unknown AWS Bedrock model type {self._underlying_provider}",
+            )
+
+        return preped
+
+    def _process_output(self, response: dict):
+        if self._underlying_provider is AWSBedrockModelProvider.ANTHROPIC:
+            return {
+                "candidates": [
+                    {
+                        "text": response.get("completion"),
+                        "metadata": {
+                            "finish_reason": "stop"
+                            if response.get("stop_reason") == "stop_sequence"
```

Comment:
Similar to https://github.com/mlflow/mlflow/pull/9598/files#r1326549811, could we factor out response translation logic from anthropic, etc. (e.g. https://github.com/mlflow/mlflow/blob/61d8622dde16578dd844ecc12668580dd81a311c/mlflow/gateway/providers/anthropic.py#L84) into a separate method and call it here?